integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/K8sEvents.java

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) 2020, Oracle Corporation and/or its affiliates.
+// Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+
+package oracle.weblogic.kubernetes.utils;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.Callable;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import io.kubernetes.client.openapi.ApiException;
+import io.kubernetes.client.openapi.models.V1Event;
+import io.kubernetes.client.util.Yaml;
+import oracle.weblogic.kubernetes.ItKubernetesEvents;
+import oracle.weblogic.kubernetes.TestConstants;
+import oracle.weblogic.kubernetes.actions.TestActions;
+import oracle.weblogic.kubernetes.actions.impl.primitive.Kubernetes;
+import oracle.weblogic.kubernetes.logging.LoggingFacade;
+import org.joda.time.DateTime;
+
+import static oracle.weblogic.kubernetes.TestConstants.OPERATOR_RELEASE_NAME;
+import static oracle.weblogic.kubernetes.utils.ThreadSafeLogger.getLogger;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/**
+ * Helper class for Kubernetes Events checking.
+ */
+public class K8sEvents {
+
+  private static final LoggingFacade logger = getLogger();
+
+  /**
+   * Check if a given event is logged by the operator.
+   *
+   * @param opNamespace namespace in which the operator is running
+   * @param domainNamespace namespace in which the domain exists
+   * @param domainUid UID of the domain
+   * @param reason event to check for Created, Changed, deleted, processing etc
+   * @param type type of event, Normal of Warning
+   * @param timestamp the timestamp after which to see events
+   */
+  public static Callable<Boolean> checkDomainEvent(
+      String opNamespace, String domainNamespace, String domainUid, String reason, String type, DateTime timestamp) {
+    return () -> {
+      logger.info("Verifying {0} event is logged by the operator in domain namespace {1}", reason, domainNamespace);
+      try {
+        List<V1Event> events = Kubernetes.listNamespacedEvents(domainNamespace);
+        for (V1Event event : events) {
+          if (event.getReason().contains(reason)
+              && event.getMetadata().getCreationTimestamp().isAfter(timestamp.getMillis())) {
+            logger.info(Yaml.dump(event));
+            verifyOperatorDetails(event, opNamespace, domainUid);
+            //verify reason
+            logger.info("Verifying domain event {0}", reason);
+            assertTrue(event.getReason().equals(reason));
+            //verify messages
+            logger.info("Verifying domain event message {0}", getDomainEventMessage(reason, domainUid));
+            assertTrue(event.getMessage().contains(getDomainEventMessage(reason, domainUid)));
```

Comment:
I'm concerned about having the integration test look for a specific message text. This seems too tightly coupled. What do others think? Would it be enough to have the test look for a reason code?

Reply:
I am okay with just checking the reason code; the unit tests check the messages (and other fields) already.

Reply:
removed the message checking